### PR TITLE
Clarify duration/end_date_time usage

### LINKED ISSUE #29787: Incorrect details for service calendar.list_events
### Feedback

While working with calendar.list_events recently, I found that Home Assistant expects a value for either `end_date_time` or `duration` when a call is made to the service. However, the documentation for the Calendar integration has all attributes for the service marked as optional:

list_events

This was the error returned by HA:

```
Error Message: must contain at least one of end_date_time, duration.. Got
{
    'type': 'execute_script',
    'sequence': [
        {
            'service': 'calendar.list_events',
            'target': {
                'entity_id': [
                    calendar.my_calendar
                ]             
            },
            'response_variable': 'agenda'
        },
        {
            'stop': 'done',
            'response_variable': 'agenda'
        }     
    ],
    'id': 34
}
```

### URL

https://www.home-assistant.io/integrations/calendar/

### Version

2023.11.1

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by FletcherAU**
## Proposed change

Add a note to `duration` that it's mutually exclusive with `end_date_time`. (the latter of which already had this note)

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #29787

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by c0ffeeca7** [comment]
@FletcherAU, you marked this PR as draft a while ago.   Is it ready for review? Or are you still working on it?

**Comment by FletcherAU** [comment]
Ideally this change would include a clarification in the `Optional` column but I'm not sure whether that is in keeping with the standard set by other pages. I was holding off until there was clarification in either #29787 or #dev-documentation :)

File: source/_integrations/calendar.markdown [hunk 1]
```diff
@@ -201,7 +201,7 @@ with calendar events within a date range.
 | ---------------------- | -------- | ----------- | --------|
 | `start_date_time` | yes | Return active events after this time (exclusive). When not set, defaults to now. | 2019-03-10 20:00:00
 | `end_date_time` | yes | Return active events before this time (exclusive). Cannot be used with 'duration'. | 2019-03-10 23:00:00
-| `duration` | yes | Return active events from start_date_time until the specified duration. | `days: 2`
+| `duration` | yes | Return active events from start_date_time until the specified duration. Cannot be used with 'end_date_time'. | `days: 2`
```
Comment (c0ffeeca7) [inline source/_integrations/calendar.markdown]:
```suggestion
| `duration` | yes | Return active events from start_date_time until the specified duration. Cannot be used with `end_date_time`. You must specify either `duration` or `end_date_time`. | `days: 2`
```

File: source/_integrations/calendar.markdown [hunk 2]
```diff
@@ -201,7 +201,7 @@ with calendar events within a date range.
 | ---------------------- | -------- | ----------- | --------|
 | `start_date_time` | yes | Return active events after this time (exclusive). When not set, defaults to now. | 2019-03-10 20:00:00
 | `end_date_time` | yes | Return active events before this time (exclusive). Cannot be used with 'duration'. | 2019-03-10 23:00:00
```
Comment (c0ffeeca7) [inline source/_integrations/calendar.markdown]:
```suggestion
| `end_date_time` | yes | Return active events before this time (exclusive). Cannot be used with `duration`. You must specify either `end_date_time` or `duration`.| 2019-03-10 23:00:00
```

File: source/_integrations/calendar.markdown [hunk 3]
```diff
@@ -200,8 +200,8 @@ with calendar events within a date range.
 | Service data attribute | Optional | Description | Example |
 | ---------------------- | -------- | ----------- | --------|
 | `start_date_time` | yes | Return active events after this time (exclusive). When not set, defaults to now. | 2019-03-10 20:00:00
-| `end_date_time` | yes | Return active events before this time (exclusive). Cannot be used with 'duration'. | 2019-03-10 23:00:00
-| `duration` | yes | Return active events from start_date_time until the specified duration. | `days: 2`
+| `end_date_time` | yes | Return active events before this time (exclusive). Cannot be used with `duration`. You must specify either `end_date_time` or `duration`.| 2019-03-10 23:00:00
+| `duration` | yes | Return active events from start_date_time until the specified duration. Cannot be used with `end_date_time`. You must specify either `duration` or `end_date_time`. | `days: 2`
```
Comment (c0ffeeca7) [inline source/_integrations/calendar.markdown]:
```suggestion
| `duration` | yes | Return active events from `start_date_time` until the specified duration. Cannot be used with `end_date_time`. You must specify either `duration` or `end_date_time`. | `days: 2`
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @FletcherAU :+1:

**Pull request merged**